packages/services/src/competition.service.ts

```diff
@@ -2748,60 +2739,42 @@ export class CompetitionService {
     pagingParams: PagingParams;
   }): Promise<EnrichedCompetitionsData> {
     try {
-      // Get competitions
-      const { competitions, total } = await this.getCompetitions(
-        params.status,
-        params.pagingParams,
-      );
-
-      // Enrich competitions with trading constraint information
-      let enrichedCompetitions = competitions;
-      const competitionIds = competitions.map((c) => c.id);
-
-      const enrichmentData =
-        await this.competitionRepo.getEnrichedCompetitions(competitionIds);
-
-      // Create lookup maps for efficient access
-      const enrichmentMap = new Map(
-        enrichmentData.map((data) => [data.competitionId, data]),
-      );
-
-      enrichedCompetitions = competitions.map((competition) => {
-        const enrichment = enrichmentMap.get(competition.id);
-        if (!enrichment) {
-          throw new ApiError(500, "invalid competition state");
-        }
+      const { competitions, total } = await this.competitionRepo.findByStatus({
+        status: params.status,
+        params: params.pagingParams,
+      });
 
-        const tradingConstraints = {
-          minimumPairAgeHours: enrichment.minimumPairAgeHours,
-          minimum24hVolumeUsd: enrichment.minimum24hVolumeUsd,
-          minimumLiquidityUsd: enrichment.minimumLiquidityUsd,
-          minimumFdvUsd: enrichment.minimumFdvUsd,
-          minTradesPerDay: enrichment.minTradesPerDay,
-        };
+      const enrichedCompetitions = competitions.map((competition) => {
```

Comment:
let me know if you think it'd be better to push this transformation into the repository. Would help clean up the ugly return type specification on this service function, but would make the repository be worrying about the shape of the return data which feels slightly layer-violation-y?

Reply:
yeah that's fair.  I guess those inferred types like `Awaited` and `RouterOutputs` have been growing on me.  It seems my IDE is smart enough to figure out the actual response shape, which is the real important thing